### ISSUE TITLE
Chrome driver.

### ISSUE THREAD

**Issue by ill-yes** [assafelovic/gpt-researcher#127]
I am currently having problems with the Chrome driver. 

In the readme I read that downgrading Chrome would help. However, I have to say that from a security point of view this is not an option for me and on the other hand my company forbids version downgrade.

Is there a possibility to consider other fixes? Maybe in the Docker container itself?

```
gpt-researcher-gpt-researcher-1  | ERROR:    Exception in ASGI application
gpt-researcher-gpt-researcher-1  | Traceback (most recent call last):
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/websockets/wsproto_impl.py", line 234, in run_asgi
gpt-researcher-gpt-researcher-1  |     result = await self.app(self.scope, self.receive, self.send)
gpt-researcher-gpt-researcher-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
gpt-researcher-gpt-researcher-1  |     return await self.app(scope, receive, send)
gpt-researcher-gpt-researcher-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 289, in __call__
gpt-researcher-gpt-researcher-1  |     await super().__call__(scope, receive, send)
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 122, in __call__
gpt-researcher-gpt-researcher-1  |     await self.middleware_stack(scope, receive, send)
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 149, in __call__
gpt-researcher-gpt-researcher-1  |     await self.app(scope, receive, send)
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
gpt-researcher-gpt-researcher-1  |     raise exc
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
gpt-researcher-gpt-researcher-1  |     await self.app(scope, receive, sender)
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
gpt-researcher-gpt-researcher-1  |     raise e
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
gpt-researcher-gpt-researcher-1  |     await self.app(scope, receive, send)
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 718, in __call__
gpt-researcher-gpt-researcher-1  |     await route.handle(scope, receive, send)
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 341, in handle
gpt-researcher-gpt-researcher-1  |     await self.app(scope, receive, send)
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 82, in app
gpt-researcher-gpt-researcher-1  |     await func(session)
gpt-researcher-gpt-researcher-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 324, in app
gpt-researcher-gpt-researcher-1  |     await dependant.call(**values)
gpt-researcher-gpt-researcher-1  |   File "/usr/src/app/main.py", line 60, in websocket_endpoint
gpt-researcher-gpt-researcher-1  |     await manager.start_streaming(task, report_type, agent, agent_role_prompt, websocket)
gpt-researcher-gpt-researcher-1  |   File "/usr/src/app/agent/run.py", line 38, in start_streaming
gpt-researcher-gpt-researcher-1  |     report, path = await run_agent(task, report_type, agent, agent_role_prompt, websocket)
gpt-researcher-gpt-researcher-1  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gpt-researcher-gpt-researcher-1  |   File "/usr/src/app/agent/run.py", line 50, in run_agent
gpt-researcher-gpt-researcher-1  |     await assistant.conduct_research()
gpt-researcher-gpt-researcher-1  |   File "/usr/src/app/agent/research_agent.py", line 143, in conduct_research
gpt-researcher-gpt-researcher-1  |     research_result = await self.run_search_summary(query)
gpt-researcher-gpt-researcher-1  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gpt-researcher-gpt-researcher-1  |   File "/usr/src/app/agent/research_agent.py", line 128, in run_search_summary
gpt-researcher-gpt-researcher-1  |     os.makedirs(os.path.dirname(f"./outputs/{self.directory_name}/research-{query}.txt"), exist_ok=True)
gpt-researcher-gpt-researcher-1  |   File "<frozen os>", line 225, in makedirs
gpt-researcher-gpt-researcher-1  | PermissionError: [Errno 13] Permission denied: './outputs/267a4451-ea58-4684-a385-ed11bd8ef6f7'
gpt-researcher-gpt-researcher-1  | INFO:     ('192.168.48.1', 38550) - "WebSocket /ws" [accepted]
gpt-researcher-gpt-researcher-1  | An error occurred while processing the url https://digital-strategy.ec.europa.eu/de/policies/european-approach-artificial-intelligence: Message: unknown error: Chrome failed to start: exited abnormally.
gpt-researcher-gpt-researcher-1  |   (chrome not reachable)
gpt-researcher-gpt-researcher-1  |   (The process started from chrome location /home/gpt-researcher/.cache/selenium/chrome/linux64/115.0.5790.170/chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
gpt-researcher-gpt-researcher-1  | Stacktrace:
gpt-researcher-gpt-researcher-1  | #0 0xaaaad8ab4e6c <unknown>
gpt-researcher-gpt-researcher-1  | #1 0xaaaad88095b4 <unknown>
gpt-researcher-gpt-researcher-1  | #2 0xaaaad882f800 <unknown>
gpt-researcher-gpt-researcher-1  | #3 0xaaaad882c608 <unknown>
gpt-researcher-gpt-researcher-1  | #4 0xaaaad886da74 <unknown>
gpt-researcher-gpt-researcher-1  | #5 0xaaaad8835950 <unknown>
gpt-researcher-gpt-researcher-1  | #6 0xaaaad8836bdc <unknown>
gpt-researcher-gpt-researcher-1  | #7 0xaaaad8a81c48 <unknown>
gpt-researcher-gpt-researcher-1  | #8 0xaaaad8a852e4 <unknown>
gpt-researcher-gpt-researcher-1  | #9 0xaaaad8a84e80 <unknown>
gpt-researcher-gpt-researcher-1  | #10 0xaaaad8a8be90 <unknown>
gpt-researcher-gpt-researcher-1  | #11 0xaaaad8a859ec <unknown>
gpt-researcher-gpt-researcher-1  | #12 0xaaaad8a5f7e4 <unknown>
gpt-researcher-gpt-researcher-1  | #13 0xaaaad8a9f114 <unknown>
gpt-researcher-gpt-researcher-1  | #14 0xaaaad8a9f2dc <unknown>
gpt-researcher-gpt-researcher-1  | #15 0xaaaad8aad468 <unknown>
gpt-researcher-gpt-researcher-1  | #16 0xffffac6fb648 start_thread
gpt-researcher-gpt-researcher-1  | #17 0xffffabea4fdc <unknown>
gpt-researcher-gpt-researcher-1  | 
gpt-researcher-gpt-researcher-1  | An error occurred while processing the url https://www.europarl.europa.eu/news/en/headlines/society/20230601STO93804/eu-ai-act-first-regulation-on-artificial-intelligence: Message: unknown error: Chrome failed to start: exited abnormally.
gpt-researcher-gpt-researcher-1  |   (chrome not reachable)
gpt-researcher-gpt-researcher-1  |   (The process started from chrome location /home/gpt-researcher/.cache/selenium/chrome/linux64/115.0.5790.170/chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
gpt-researcher-gpt-researcher-1  | Stacktrace:
gpt-researcher-gpt-researcher-1  | #0 0xaaaac5d84e6c <unknown>
gpt-researcher-gpt-researcher-1  | #1 0xaaaac5ad95b4 <unknown>
```

**Comment by smach** [comment]
It works fine with Firefox. That seems a better idea than downgrading an existing version of Chrome for anyone who has the option of also using Firefox.

**Comment by ill-yes** [comment]
How did u set Firefox as browser for this?

**Comment by tomseidel** [comment]
Fixing the user-agent param worked for me I think. Maybe a windows formating issue.
# options.add_argument(CFG.user_agent)
options.add_argument(f"user-agent={CFG.user_agent}")

**Comment by smach** [comment]
@ill-yes You should be able to switch your default browser from Chrome to Firefox in your computer's settings. I didn't want to do that, I just copied and pasted the URL from Chrome into Firefox.

**Comment by assafelovic** [comment]
@tomseidel thank you for the suggestion we've just merged that #159
